test/AutoRest.Shared.Tests/TypeFormatterTests.cs

```diff
@@ -96,5 +106,31 @@ public void FormatsTimeSpanAsJson(string format, TimeSpan duration, string expec
             Assert.AreEqual(expected, formatted.ToString());
             Assert.AreEqual(duration, formatted.GetTimeSpan(format));
         }
+
+        [TestCase(null, null, "null")]
+        [TestCase(null, "str", "str")]
+        [TestCase(null, true, "true")]
+        [TestCase(null, false, "false")]
+        [TestCase(null, 42, "42")]
+        [TestCase(null, -42, "-42")]
+        [TestCase(null, 3.14f, "3.14")]
+        [TestCase(null, -3.14f, "-3.14")]
+        [TestCase(null, 3.14, "3.14")]
+        [TestCase(null, -3.14, "-3.14")]
+        [TestCase(null, 299792458L, "299792458")]
+        [TestCase(null, -299792458L, "-299792458")]
+        [TestCase("D", new byte[] { 1, 2, 3 }, "AQID")]
+        [TestCase("U", new byte[] { 4, 5, 6 }, "BAUG")]
+        [TestCase(null, new string[] { "a", "b" }, "a,b")]
+        [TestCaseSource(nameof(DateTimeOffsetCases))]
+        [TestCaseSource(nameof(TimeSpanWithoutFormateCases))]
+        [TestCaseSource(nameof(TimeSpanCases))]
+        [TestCaseSource(nameof(GuidCases))]
+        public void ValidateConvertToString(string? format, object? value, string expected)
```

Comment:
Is there a new case for the new format you added in this PR?

Reply:
No. The same test cases as before.

Reply:
Well I mean we should add one for the new format.

Reply:
Hello @ArcturusZhang I don't add new format in the `ConvertToString` function, I just move it from `RequestUriBuilderExtensions.cs` to `TypeFormatter.cs`, so that it can be shared by others.